xled/realtime.py

```diff
@@ -0,0 +1,68 @@
+# -*- coding: utf-8 -*-
+
+"""
+xled.realtime
+~~~~~~~~~~~~~
+
+Functions to support the realtime mode of the device.
+"""
+
+from __future__ import absolute_import
+
+import base64
+import math
+import socket
+
+from xled.control import ControlInterface
+
+#: UDP port to send realtime frames to
+REALTIME_UDP_PORT_NUMBER = 7777
+
+
+class RealtimeChannel(object):
+    """
+    Main interface to send realtime frames to device.
+
+    :param control: An activated ControlInterface for the device to control
+    :param int leds_number: the number of leds in a frame
+    :param int bytes_per_led: the number of bytes per led (3 or 4)
+    """
+
+    def __init__(self, control, leds_number, bytes_per_led):
+        self.control = control
+        self.leds_number = leds_number
+        self.bytes_per_led = bytes_per_led
+
+    def start_realtime(self):
```

Comment:
I'm not sure about this interface / API. I understand that it might be useful to have it here.

On the other hand if we have start here we would need a stop here - otherwise we end up with a device that stops getting new data and displays just last frame for couple of seconds until it timeouts.

And how would the stop look?
a) it could be either 'movie', 'effect' or 'playlist' depending on the desirable state which would be decided on a caller
b) or it could store state before realtime start end switch to original one at the end.

In the case of a) one this would just duplicate corresponding methods from control one. Or leak underlying API. In the end - do we need `start_realtime()` if the caller could manage to do that on their own?

b) might be interface that is easier for a caller. But in the end it would need much more logic for a class that is indended for RealtimeChannel. So I guess in that case it might be better to have all the logic in higher level interface where one would not call start nor stop and only send data. Maybe even define Context manager (with `with` keyword).